library/lib/core/application.cpp

```diff
@@ -161,10 +163,45 @@ bool Application::mainLoop()
 
     // Input
     ControllerState controllerState = {};
+    RawTouch rawTouch               = {};
 
     InputManager* inputManager = Application::platform->getInputManager();
+    inputManager->updateTouchState(&rawTouch);
     inputManager->updateControllerState(&controllerState);
 
+    static Touch oldTouch;
+    Touch touchState = InputManager::computeTouchState(rawTouch, oldTouch);
+    oldTouch         = touchState;
+
+    // Touch controller events
+    switch (touchState.phase)
+    {
+        case TouchPhase::START:
+            Logger::debug("Touched at X: " + std::to_string(touchState.position.x) + ", Y: " + std::to_string(touchState.position.y));
+            Application::setInputType(InputType::TOUCH);
+
+            // Search for first responder, which will be the root of recognition tree
+            firstResponder = Application::activitiesStack[Application::activitiesStack.size() - 1]
+                                 ->getContentView()
+                                 ->hitTest(touchState.position);
+            break;
+        case TouchPhase::NONE:
+            firstResponder = nullptr;
+            break;
+        default:
+            break;
+    }
+
+    if (firstResponder)
+    {
+        if (firstResponder->gestureRecognizerRequest(touchState, firstResponder))
+        {
+            // Play touch sound with random pitch
+            float pitch = (rand() % 10) / 10.0f + 1.0f;
+            Application::getAudioPlayer()->play(SOUND_TOUCH, pitch);
```

Comment:
There are multiple sounds to play depending on the view, just like when using a controller

For instance the clicking sound on a sidebar item is different than the clicking sound on a button. So each view should provide a "sound to play when touched", maybe depending on the touch event too

You should open the HOS settings app and touch it everywhere to see how it behaves, it will be better than my poor explanations lol

Reply:
I think you don't get what that sound is. This sound is "nothing to touch", it's the same everywhere you click on unclickable part of UI. 

I attached screenshots where you can see the parts (not strikethrough) where this sound generated, if you'll check, I fully reproduced HOS sound behaviour, where selected sideview tab not selecting again and produce that "nothing" sound, or disabled\no action button also produce that sound.

![IMG_1151-min](https://user-images.githubusercontent.com/9553519/112226093-c8396e00-8c3e-11eb-8946-b061a35317b0.jpg)
![IMG_7745-min](https://user-images.githubusercontent.com/9553519/112226107-ccfe2200-8c3e-11eb-948f-6f8e62cdfc0e.jpg)